### ISSUE TITLE
Logging grad norm

### ISSUE THREAD

**Issue by DavidSlayback** [deepmind/optax#326]
A quick question:

In PyTorch, I found it useful to be able to log the original norm when applying gradient clipping to get an idea of the scale of the updates being attempted (e.g., `original_norm = nn.utils.clip_grad_norm_(agent.parameters(), args.max_grad_norm)`

Is there an equivalent in Optax? Currently, I have norm clipping as part of the chain, but it's not clear how I would extract the original gradient norm value

**Comment by hbq1** [comment]
Hi David, would it be possible to split your chain of transformations into 2 parts and use the output of the 1st part for logging?

**Comment by hbq1** [comment]
Closing this issue for now, but feel free to re-open if any questions remain.
